File: WordSwell.Frontend/src/Faculty/Backend/BoardCont/PostEditApplyCallModel.ts
import { FileItemModel } from '@/Faculty/Backend/FileDb/FileItemModel';

/** 게시물 수정 요청 모델 */
export interface PostEditApplyCallModel 
{
    /** 게시판 고유 번호 */
    idBoard: number,
    /** 게시물 고유 번호 */
    idBoardPost: number,
    /** 제목 */
    Title: string,
    /** 비밀번호 - 비회원이 글쓴 경우 넣는다. */
    Password: string,
    /** 게시물 내용 */
    Contents: string,
    /** 첨부 파일 리스트 */
    FileList?: FileItemModel[],
}

File: WordSwell.ApiModels/BoardCont/PostEditApplyCallModel.cs
using WordSwell.ApiModels.FileDb;

namespace WordSwell.ApiModels.BoardCont;

/// <summary>
/// 게시물 수정 요청 모델
/// </summary>
public class PostEditApplyCallModel
{
    /// <summary>
    /// 게시판 고유 번호
    /// </summary>
    public long idBoard { get; set; }

    /// <summary>
    /// 게시물 고유 번호
    /// </summary>
    public long idBoardPost { get; set; }


    /// <summary>
    /// 제목
    /// </summary>
    public string? Title { get; set; } = string.Empty;

    /// <summary>
    /// 비밀번호 - 비회원이 글쓴 경우 넣는다.
    /// </summary>
    public string Password { get; set; } = string.Empty;

    /// <summary>
    /// 게시물 내용
    /// </summary>
    public string? Contents { get; set; } = string.Empty;

    /// <summary>
    /// 첨부 파일 리스트
    /// </summary>
    public List<FileItemModel>? FileList { get; set; }
}
